### ISSUE TITLE
https://github.com/mckaywrigley/chatbot-ui   Compatibility!!!!!!!!!!!!

### ISSUE THREAD

**Issue by PaulShroom** [C-Nedelcu/talk-to-chatgpt#109]
Hi C-Nedelcu,

Please can you make it work with this? https://github.com/mckaywrigley/chatbot-ui

I did some testing and it dose write text but not in the correct place, can it be  adapted?

All the best Paul

**Comment by C-Nedelcu** [comment]
Hello,

It seems like the project you are pointing to is a self hosted solution.
The issue that this causes is that I cannot target specific sites where the Talk-to-chatGPT widget should be enabled.

The way a browser extension works is that you must indicate on which URLs the extension will be active. Currently, it is only active for chat.openai.com. But since "chatbot-ui" is a self hosted project, I cannot know the URLs, so I cannot activate it anywhere in particular.

The only way this could work is if you do this:
- download Talk-to-ChatGPT source
- change the URL in the manifest.json
- install the modified unpacked extension manually in your browser
- adjust the code to make it work with the chatbot-ui UI

However this is a long shot and it probably wouldn't work great. You would also have to host chatbot-UI on HTTPS (SSL)  because in clear-text HTTP the microphone/speech recognition and TTS APIs wont trigger.

In short, you'd better ask the chatbot-UI team to re-use some of my code to achieve some sort of SR/TTS functionality, this would be a much simpler way to achieve what you're suggesting.
